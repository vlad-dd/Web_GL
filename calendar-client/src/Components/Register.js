import React from 'react';
import { Link } from 'react-router-dom';



const Register = (props) => {
    return (
        <div>
             <img class="image" src="http://www.cfdesigner.com/codepen/rocket-page-logo.png" alt="rocket-image" />
            <form className="register_form d-table mx-auto bg-light">
                <h2>Register</h2>
                <div className="form-group">
                    <label for="exampleInputEmail1">Email address</label>
                    <input onChange={props.updateInput} type="email" className="form-control" name="email" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email" />
                    <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small>
                </div>
                <div className="form-group">
                    <label for="register_firstname">First name</label>
                    <input onChange={props.updateInput} className="form-control" name="firstname" id="register_firstname" placeholder="Enter first name" />
                </div>
                <div className="form-group">
                    <label for="register_lastname">Last name</label>
                    <input onChange={props.updateInput} className="form-control" name="lastname" id="register_lastname" placeholder="Enter last name" />
                </div>
                <div className="form-group">
                    <label for="exampleInputPassword1">Password</label>
                    <input onChange={props.updateInput} type="password" className="form-control" name="password" id="exampleInputPassword1" placeholder="Password" />
                </div>

                <div className="form-group">
                    <label for="exampleInputPassword2">Retype password</label>
                    <input onChange={props.updateInput} type="password" className="form-control" name="rPassword" id="exampleInputPassword2" placeholder="Retype password" />
                </div>

                <Link to='/login' className="d-block">Already registered?</Link>


                <button type="submit" class="btn btn-primary">Submit</button>
            </form>
        </div>)
}

export default Register;