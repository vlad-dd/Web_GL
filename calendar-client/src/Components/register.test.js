import React from 'react';
import Register from './Register'
// import { shallow, render, mount, configure } from 'enzyme';



test('the input data is correct ', () => {
    expect(Register).toMatchSnapshot();
});

describe('InputWithValidation tests', () => {
    it('should render component with initial state', () => {
        const props = {
            handleClick: jest.fn()
        }
        const wrapper = shallow(<Register  {...props}/>);
        expect(wrapper.state('value').toBe(''))
        expect(wrapper.find('input').length).toBe(0)
        expect(wrapper.find('button').length).toBe(0)

    })
});

test('map calls its argument with a non-null argument', () => {
    const mock = jest.fn();
    [1].map(x => mock(x));
    expect(mock).toBeCalledWith(expect.anything());
});

it("saves on submit", function(){
    Register.done = jest.genMockFunction();
    Register.insideDone = jest.genMockFunction();
  TestUtils.Simulate.submit(form-group);
  expect(Register.done).toBeCalled(); 
  expect(Register.insideDone).toBeCalled(); 
});

// const wrapper = shallow(<Register {...props} />);
// wrapper.find('input').simulate('change', {target: {value: 'not an email'}});
// expect(wrapper.find('button').length).toBe(1);

// wrapper.find('input').simulate('change', {target: {value: 'test@test.com'}});
// expect(wrapper.find('button').length).toBe(1);