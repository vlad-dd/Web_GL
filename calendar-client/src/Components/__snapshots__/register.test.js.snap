// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the input data is correct  1`] = `[Function]`;
